shoes-swt/lib/shoes/swt/redrawing_aspect.rb

```diff
@@ -79,7 +79,10 @@ def add_redraws
       end
 
       def update_gui
-        app.flush unless app.disposed?
+        unless app.disposed?
+          app.flush
+          app.redraw
```

Comment:
huh, this doesn't seem right as we tell it to redraw the whole UI? IIRC flush was also just "ok now really do those redraws I been telling you to do" so order also seems a bit weird. It's been a long time, I'm not sure this is the appropriate fix, although I can see how it fixes things :)

Reply:
So I admit that this did strike me as a heavy hammer. `flush`, though, is [only telling the UI to do it's layout again](https://github.com/shoes/shoes4/blob/master/shoes-swt/lib/shoes/swt/app.rb#L120). That doesn't seem to always correctly trigger the redrawing area in SWT, especially when our shown region gets bigger as a result of a change (like the animation here).

I'm going to give this more reading and some solid testing through the heavier samples before moving forward. Will report back.

Reply:
My idea/hope would be that if we correctly redraw all areas/objects that are moved during the animation then we shouldn't need the whole app redraw... of course can't totally say :)

Reply:
Yeah, it does seem like something is probably being missed at the lower levels there. Looking it over, I grow a bit suspicious that maybe our redrawing aspect hook is missing something it needs on resizes for elements. I would have thought `_position` would handle it, but maybe not. Will investigate.

Reply:
One thing that springs to my mind, we only redraw the new location but maybe not the old position, but not sure how to get that :scream: